### PR TITLE
[Test] enable tests under protocol and serialization folders

### PR DESCRIPTION
**Pull request by jikunshang**
issue #17 and issue #18

### CODE REVIEW

File: cpp/src/protocol/tests/SplitTest.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
+
```
Comment (winningsix) [inline cpp/src/protocol/tests/SplitTest.cpp:1]: nit: license header.

File: cpp/src/protocol/tests/SplitTest.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      EXPECT_TRUE(veloxSplit.hasConnectorSplit());
+      auto ptr =
+          std::dynamic_pointer_cast<facebook::velox::connector::tpch::TpchConnectorSplit>(
+              veloxSplit.connectorSplit);
+      EXPECT_TRUE(ptr != nullptr);
+    }
+  }
+}
+
+TEST_F(ProtocolTest, RemoteConnectorSplit) {
```
Comment (winningsix) [inline cpp/src/protocol/tests/SplitTest.cpp:44]: Can we implement a helper method given it's quite similar as https://github.com/oap-project/Gluten-Trino/pull/25/files#diff-74c9a3fea84898739c46a15531b91fc89a6724a56dbea468f7eada1d93d6882bR27-R41
?

File: cpp/src/protocol/tests/SplitTest.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+int main(int argc, char** argv) {
+  testing::InitGoogleTest(&argc, argv);
+
+  int err{0};
+  err = RUN_ALL_TESTS();
+
+  return err;
+  err = RUN_ALL_TESTS();
```
Comment (winningsix) [inline cpp/src/protocol/tests/SplitTest.cpp:87]: hmm, who called twice? And it's after return statement.

**Review by winningsix — APPROVED** [review]

File: cpp/src/serialization/tests/SerdeTest.cpp [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+  facebook::velox::test::assertEqualVectors(deserialized, c);
+}
+
+int main(int argc, char** argv) {
+  testing::InitGoogleTest(&argc, argv);
+
+  int err{0};
+  err = RUN_ALL_TESTS();
+
+  return err;
```
Comment (wuxueyang96) [inline cpp/src/serialization/tests/SerdeTest.cpp:197]: I don't understand what it means here. Why should we run all tests twice?

**Review by wuxueyang96 — APPROVED** [review]

**Pull request merged**